File: project/src/main/java/br/com/project/repositories/TagRepository.java
package br.com.project.repositories;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import br.com.project.models.Tag;
import br.com.project.models.User;

@Repository
public interface TagRepository extends JpaRepository<Tag, UUID> {
	boolean existsByTagNameContainingIgnoreCaseAndUser(String tagName, User user);
	List<Tag> findAllByUser(Optional<User> findById);
	Tag findByTagNameAndUser(String tagName, User user);
}

File: src/main/kotlin/br/com/project/projetoIntegrador/repositories/TagRepository.kt
package br.com.project.projetoIntegrador.repositories

import br.com.project.projetoIntegrador.models.Tag
import br.com.project.projetoIntegrador.models.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface TagRepository : JpaRepository<Tag, UUID> {

    fun existsByTagNameContainingIgnoreCaseAndUser(tagName: String, user: User): Boolean
    fun findAllByUser(findById: Optional<User>): List<Tag>
    fun findByTagNameAndUser(tagName: String?, user: User): Tag
}